packages/dd-trace/src/appsec/index.js

```diff
@@ -117,34 +121,49 @@ function incomingHttpEndTranslator ({ req, res }) {
     payload[addresses.HTTP_INCOMING_COOKIES] = req.cookies
   }
 
+  // TODO: no need to analyze it if it was already done by the body-parser hook
+  if (req.query !== undefined && req.query !== null) {
+    payload[addresses.HTTP_INCOMING_QUERY] = req.query
+  }
+
   waf.run(payload, req)
 
   waf.disposeContext(req)
 
   Reporter.finishRequest(req, res)
 }
 
-function onRequestBodyParsed ({ req, res, abortController }) {
+function onRequestBodyParsed ({ req, res, body, abortController }) {
+  if (!req) {
+    const store = storage.getStore()
+    req = store?.req
+  }
+
   const rootSpan = web.root(req)
   if (!rootSpan) return
 
-  if (req.body === undefined || req.body === null) return
+  if (body === undefined || body === null) return
 
   const results = waf.run({
-    [addresses.HTTP_INCOMING_BODY]: req.body
+    [addresses.HTTP_INCOMING_BODY]: body
   }, req)
 
   handleResults(results, req, res, rootSpan, abortController)
 }
 
-function onRequestQueryParsed ({ req, res, abortController }) {
+function onRequestQueryParsed ({ req, res, query, abortController }) {
+  if (!req) {
+    const store = storage.getStore()
+    req = store?.req
```

Comment:
same here obtaining the `query` from req?

Reply:
Same here, query is not necessarily in the request.